cpp/daal/src/algorithms/k_nearest_neighbors/bf_knn_classification_predict_kernel.h

```diff
@@ -35,12 +35,25 @@ namespace internal
 {
 using namespace daal::data_management;
 
+enum class PairwiseDistanceType
+{
+    minkowski,
+    chebychev
+};
+
+struct KernelParameter : bf_knn_classification::Parameter
+{
+    bf_knn_classification::VoteWeights voteWeights = bf_knn_classification::VoteWeights::voteUniform;
+    PairwiseDistanceType pairwiseDistance          = PairwiseDistanceType::minkowski;
+    double minkowskiDegree                         = 2.0;
+};
+
 template <typename algorithmFpType, CpuType cpu>
 class KNNClassificationPredictKernel : public daal::algorithms::Kernel
 {
 public:
     services::Status compute(const NumericTable * data, const classifier::Model * m, NumericTable * label, NumericTable * indices,
-                             NumericTable * distances, const daal::algorithms::Parameter * par);
+                             NumericTable * distances, const KernelParameter * par);
```

Comment:
Why do you pass it as pointer, not reference?

Reply:
Because it was originally passed as a pointer, less code changes by the way.